pika/adapters/select_connection.py

```diff
@@ -367,11 +373,11 @@ def _process_fd_events(self, fd_event_map, write_only):
                 handler = self._fd_handlers[fileno]
                 handler(fileno, events, write_only=write_only)
 
-    def flush_outbound(self):
+    def _flush_outbound(self):
         """Call the state manager who will figure out that we need to write
             then call the poller's poll function to force it to process events.
         """
-        self._()
+        self._manage_event_state()
```

Comment:
Err, _manage_event_state() is defined is BaseConnection()

On Thu, 7 May 2015 at 08:45 vitaly-krugl <notifications@github.com> wrote:

> In pika/adapters/select_connection.py
> <https://github.com/pika/pika/pull/559#discussion_r29830763>:
>
> >          """Call the state manager who will figure out that we need to write
> >              then call the poller's poll function to force it to process events.
> >          """
> > -        self._()
> > +        self._manage_event_state()
>
> @wjps <https://github.com/wjps>, this will crash, because SelectPoller
> doesn't have _manage_event_state as a member.
>
> If confirmed, this code path could also benefit from a test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pika/pika/pull/559/files#r29830763>.
>